### ISSUE TITLE
Structure DDC code such that "step over next function" in the browser Debugger with a throw doesn't enter SDK code

### ISSUE THREAD

**Issue by natebosch** [dart-lang/sdk#34029]
Not sure how much control we have here, if this is infeasible please close.

See https://github.com/dart-lang/build/issues/1690

I think the request is that when you "step over" a line a code that throws with no catch, the stack trace is printed and we don't step in to SDK code. I don't know the behavior of this situation in vanilla JS in the browser or whether this behavior is caused by some peculiarity of how we're structuring the code.

**Comment by alan-knight** [comment]
TL;DR - hit f8, not f10. 

Also, this will behave as desired if you blackbox the SDK. Add

/dart_sdk.js\b
/ng_zone.dart$
/stack_zone_specification.dart$

We are working on a debugging tool that includes this and other options as the default, but it's still in pretty early stages.  https://github.com/dart-lang/devtools-frontend

What's happening is that it's not really the case that you're stepping over a line of code with no catch. There is a finally clause in SDK code that invokes this, and the step is taking you to the next the line. The main program is async, which means it's getting rewritten to handle that behavior, and that introduces zones and handling.

But more simply, just hit F8 when you get into code you don't recognize, and it should proceed to the end, or at least to the next exception handler.
